cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -1200,6 +1220,15 @@ private String rawRebalanceAnnotation(KafkaRebalance kafkaRebalance) {
 
     }
 
+    private Condition validateAnnotation(KafkaRebalanceState state, KafkaRebalanceAnnotation rebalanceAnnotation) {
+        Condition condition = null;
+        if (!KafkaRebalanceState.Ready.validateAnnotation(rebalanceAnnotation)) {
+            condition =  StatusUtils.buildWarningCondition("InvalidAnnotation", "Wrong annotation value: " + rebalanceAnnotation
+                    + ". The rebalance resource supports the following annotations in `Ready` state: " + state.validAnnotationString());
+        }
+        return condition;
+    }
```

Comment:
not sure I understand this method. 
AFAIU, it looks like It should validate that the `rebalanceAnnotation` is valid against the `state` (both are provided as parameter) and then returning a warning condition in the case it was not valid. Instead, it's checking that the `rebalanceAnnotation` is valid for just the `Ready` state? Why?
This method is called in a lot of places where the `Ready` state is not involved.
Other than this, I would add Javadoc as well.

Reply:
Thanks for catching it Paolo. I might have not changed it

Reply:
Resolved this issue now